### ISSUE TITLE
alternator test_fetch_from_system_tables fails when one node is down

### ISSUE THREAD

**Issue by nyh** [scylladb/scylla#6440]
When I start a three-node Alternator cluster and then kill one of them, almost all of the `test/alternator` tests work properly - as they should (Alternator never uses CL=ALL). However, one test fails:
```
$ pytest test_system_tables.py::test_fetch_from_system_tables
...
E           ClientError: An error occurred (Internal Server Error) when calling the Scan operation (reached max retries: 3): Internal server error: exceptions::unavailable_exception (Cannot achieve consistency level for cl LOCAL_ONE. Requires 1, alive 0)
```
What is special in this test that only it fails in this case? @psarna any idea?

**Comment by psarna** [comment]
Was it one of the `system_auth` tables that returned the error? We do have warnings that with certain authenticators it's recommended to manually increase the replication factor for this keyspace, since it's 1 by default:
```cql
DESCRIBE KEYSPACE system_auth;
CREATE KEYSPACE system_auth WITH replication = {'class': 'SimpleStrategy', 'replication_factor': '1'}  AND durable_writes = true;
```
and here's an excerpt from scylla_yaml:
```yaml
# - AllowAllAuthenticator performs no checks - set it to disable authentication.
# - PasswordAuthenticator relies on username/password pairs to authenticate
#   users. It keeps usernames and hashed passwords in system_auth.credentials table.
#   Please increase system_auth keyspace replication factor if you use this authenticator.
# authenticator: AllowAllAuthenticator

```

**Comment by psarna** [comment]
Now that I think of it, perhaps if `alternator_enforce_authorization`, we should also set the replication factor of `system_auth` to something higher?

And as for the test, the simplest way would be to skip `system_auth`. The test fetches all system tables simply because it's quick and it validates all different kinds of columns, but we don't specifically need any interesting corner case provided by `system_auth`.

**Comment by nyh** [comment]
I don't think we need to change the test - system_auth should have been readable just like all the rest. If anything, it caught a real bug, because Alternator authentication needs a functioning `system_auth` table.

I will write a patch that if `alternator_enforce_authorization` and the cluster has three or more nodes, **verifies** that the RF of the `system_auth` table is >= 3. Users will have follow the instructions of https://docs.scylladb.com/operating-scylla/procedures/cluster-management/add_node_to_cluster/#prerequisites on how to change the RF of `system_auth` if they want to get correct behavior (and my test-running framework will do it too), but at least if they don't they'll get an error when trying to boot the third node - instead of successful boot. 

The better solution is to fix that `system_auth` RF mess in the first place. This is issue #1796 (and many of its duplicates created over the years) - but it's beyond the scope of this issue.

**Comment by nyh** [comment]
I started coding what I suggested above but it's ugly... How much replication do we consider enough? On multiple DCs, do we need RF>=3 in total, RF>=3 on each DC or what?
I'm starting to have second thoughts - shouldn't we solve issue #1796 once and for all, automatically, instead of explaining around it?

**Comment by psarna** [comment]
Solving #1796 would be better, but it doesn't look like we reached a consensus on what this replication factor should be. Perhaps simply setting it to 3 is still better than having a default of 1, even if it doesn't address all the problems?

**Comment by nyh** [comment]
Yes, definitely setting it to 3 would have been good enough (at least in my view...) but there's a whole can of worms involved: 3 on every DC or what (and how)? How do you properly use a RF=3 table before you have 3 nodes?

**Comment by psarna** [comment]
I'd say that even a regular 3 is better than 1, so we don't need to immediately think about DCs. And as for using the table, I suppose only the initial querying of the default user is problematic, since it uses cl=QUORUM... I suppose we could degrade this cl to LOCAL_ONE if the total number of nodes is 1? Aside from this default user, queries to auth are usually performed with LOCAL_ONE anyway.

**Comment by slivne** [comment]
We can;'t degrade the initial user reading LOCAL_ONE - its for correctness in case the password has changed.

**Comment by slivne** [comment]
I don't think this is alterantor specific - we are discussing generic solutions for this case.

**Comment by psarna** [comment]
Right, but I'm only suggesting to downgrade to LOCAL_ONE if the total number of nodes is 1, which means it's in fact as powerful as ALL - since there's only one copy of the data anyway. Not to mention that single node clusters are only seen in testing environments.
